free-3.md

```diff
@@ -9,42 +9,94 @@ good way to meet their needs.
+It makes sense to give them additional code to write because the code will be
+written in the existing code base. Furthermore, it requires language to
+communicate what exactly you want users to understand.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
-
+The motivation is from my summer internship.  I worked at (Sourcegraph)[https://sourcegraph.com], a semantic code search and global cross reference engine. One
+of the ideas thrown around was to help users understand code bases. The idea of
+a "starting point", was explicitly brought up by one of the developers, and I
+want to take this to the next level. This excites me because if done well, it
+could truly be used in every single code base and helpful to every developer
+out there.
 
 ### Domain
 _Describe the project's domain in five words._
-
+Visualizing code repositories
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
-
-
+The user would have to explicitly code where the "starting point" and trace
+points are in the repository. So, we could imagine the code in the dsl to look
+like this: 
+
+```
+import Visualizer
+
+var tracingSomeAction = new Visualizer();
+function foo() {
+	tracingSomeAction.startingPoint()
+	...
+}
+
+function bar() {
+	tracingSomeAction.tracePoint()
+}
+
+function baz() {
+	tracingSomeAction.endPoint()	
+}
+```
+
+The output would be a series of charts or trees for each Visualizer object so
+we could see where to start and the trace through the code for the action.
+ 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
-
+When the program runs, an image with links to specific parts of code would be
+generated.  There would be some CLI to run the code specifically to search for
+these user generated code snippets using the Visualization library. Errors would
+be if no starting point or end point were found, and this would be communicated
+when the CLI was run. 
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
-
+It should be easy to generate graph images. It should be difficult but possible
+to generate random art with this by creating weird visualizations. It should be
+impossible to do anything else.
 
 ### Related work
 _Are there any other DSLs in this domain? If not, describe how you know there
 aren't and conjecture why not. If so, describe them and provide links. How well 
 do they address the need? Are there any particularly admirable qualities of the
 language? Are there parts of the language you think could be improved?_
-
+There are some applicatons/DSLs in this domain. One example is (this)[http://ghv.artzub.com/].
```

Comment:
I was also thinking about the IntelliJ IDE. It doesn't give you visualizations, and I'm not sure if we would call it a language. But command-clicking on a function/variable declaration can show you where that function/variable has been used within the repo. Moreover, command-clicking on a usage of a function/variable/type can send you back to its original declaration. It's a useful feature that I find similar to your idea (but in a visual way)